torch/_inductor/codegen/cpp.py

```diff
@@ -3133,6 +3118,9 @@ def reduction_combine_vec(
             else:
                 return f"{reduction_type}_combine_vec<{cdtype}, {n_src}, {n_idx}{t_extra}>({var}, {next_value}{arg_extra})"
         elif reduction_type == "any":
+            if isinstance(next_value, CppCSEVariable):
+                assert next_value.dtype == torch.bool
+                (next_value,) = unify_mask_base_type(V.kernel.compute, (next_value,))
```

Comment:
Why only `next_value` need to cast the storage dtype?

Reply:
Make sense. But I think we should also add the `var` into the inputs (for case that `acc_vec ` changed to initialized with other data types in the future). If I remember correctly, under the CPP implementation, vecmask will check the C++ dtype and skip the cast if the target and source data type are same.

Reply:
Well, if `acc_vec`'s type doesn't match, it will fail anyway as the result of combination will be assigned back to `acc_vec`.

In addition, it seems that we just cannot apply `unify_mask_base_type` as it is usually a `str` but not `CppCSEVariable`.

https://github.com/pytorch/pytorch/blob/52a620845bff57a630e27a4438370b970136816d/torch/_inductor/codegen/cpp.py#L2753
https://github.com/pytorch/pytorch/blob/52a620845bff57a630e27a4438370b970136816d/torch/_inductor/codegen/cpp.py#L2819-L2821

Reply:
> it seems that we just cannot apply unify_mask_base_type as it is usually a str but not CppCSEVariable.

yean, current implementation only accepts `CppCSEVariable`.